Foundation/Decimal.swift

```diff
@@ -714,107 +677,131 @@ extension Decimal {
     }
 
-    public init(_ value: UInt) {
-        self.init(UInt64(value))
+    public var exponent: Int {
+        return Int(_exponent)
     }
 
-    public init(_ value: Int) {
-        self.init(Int64(value))
+    public var significand: Decimal {
+        return Decimal(
+            _exponent: 0, _length: _length, _isNegative: _isNegative, _isCompact: _isCompact,
+            _reserved: 0, _mantissa: _mantissa)
     }
 
-    @available(*, unavailable, message: "Decimal does not yet fully adopt FloatingPoint.")
-    public static var infinity: Decimal { fatalError("Decimal does not yet fully adopt FloatingPoint") }
-
-    @available(*, unavailable, message: "Decimal does not yet fully adopt FloatingPoint.")
-    public static var signalingNaN: Decimal { fatalError("Decimal does not yet fully adopt FloatingPoint") }
+    public var sign: FloatingPointSign {
+        return _isNegative == 0 ? FloatingPointSign.plus : FloatingPointSign.minus
+    }
 
-    public var isSignalingNaN: Bool {
-        return false
+    public var ulp: Decimal {
+        if !self.isFinite { return Decimal.nan }
+        return Decimal(
+            _exponent: _exponent, _length: 8, _isNegative: 0, _isCompact: 1,
+            _reserved: 0, _mantissa: (0x0001, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000))
     }
 
-    public static var nan: Decimal {
-        return quietNaN
+    public var nextUp: Decimal {
+        return self + Decimal(
+            _exponent: _exponent, _length: 1, _isNegative: 0, _isCompact: 1,
+            _reserved: 0, _mantissa: (0x0001, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000))
     }
 
-    public static var quietNaN: Decimal {
-        return Decimal(
-            _exponent: 0, _length: 0, _isNegative: 1, _isCompact: 0,
-            _reserved: 0, _mantissa: (0, 0, 0, 0, 0, 0, 0, 0))
+    public var nextDown: Decimal {
+        return self - Decimal(
+            _exponent: _exponent, _length: 1, _isNegative: 0, _isCompact: 1,
+            _reserved: 0, _mantissa: (0x0001, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000))
     }
 
+    /// The IEEE 754 "class" of this type.
     public var floatingPointClass: FloatingPointClassification {
         if _length == 0 && _isNegative == 1 {
             return .quietNaN
         } else if _length == 0 {
             return .positiveZero
         }
+        // NSDecimal does not really represent normal and subnormal in the same
+        // manner as the IEEE standard, for now we can probably claim normal for
+        // any nonzero, non-NaN values.
         if _isNegative == 1 {
             return .negativeNormal
         } else {
             return .positiveNormal
         }
     }
 
-    public var isSignMinus: Bool {
-        return _isNegative != 0
-    }
+    public var isCanonical: Bool { true }
 
-    public var isNormal: Bool {
-        return !isZero && !isInfinite && !isNaN
-    }
+    /// `true` iff `self` is negative.
```

Comment:
Is `iff` a typo here (and below)?

Reply:
No; it's the shorthand for "if and only if" synchronized verbatim from the documentation in the Darwin overlay.